File: base/code.py
# drummer2322
# corbin.mc96@gmail.com
# GCJ <year> <round> <problem>
# <date>

from sys import argv
import string

if len(argv) > 1:
	type_of_case = argv[1]+'_'
else:
	type_of_case = ''
lines = [line[:-1] for line in open(type_of_case+'input.in')]
output = open(type_of_case+'output.out','w')

digits = string.digits + string.letters

for t in range(1,int(lines.pop(0))+1):
	print t

	message = lines.pop(0);
	base = len(set(list(message)))
	base_x = "";
	base_10 = 0;
	base_matching = {};

	current_number = 1
	for i, letter in enumerate(message):
		if not letter in base_matching.keys():
			base_matching[letter] = digits[current_number] # change digits in this line to modified digits
			current_number+=1
		base_x += base_matching[letter]
	
	print base_x
	
	base_x[::-1]
	for i, letter in base_x:
		base_10 += base**i * digits.index(letter)

	output.write("Case #%i: %s\n"%(t, base_10))

output.close()

File: sheep_cpp/code.cpp
// drummer2322
// corbin.mc96@gmail.com
// GCJ <year> <round> <problem>
// <date>

#include <iostream>
#include <string>
#include <set>
#include <cstring>
#include <cstdio>

int main () 
{
	int T, tt, N, num;
	std::cin >> T;
	for (tt = 1; tt <= T; tt++) {
		std::cerr << tt << std::endl;

		std::cin >> N;
		num = N;

		bool digits[10] = {false};
		bool done=false;

		for (int i=1; i<100001; i++) {
			char number[100];
			sprintf(number, "%d", N*i);
			for (int j=0; j<strlen(number); j++) {
				digits[number[j] - '0'] = true;
			}
			done = false;
			for (int j=0; j<10; j++) {
				if (!digits[j]) {
					done = true;
					break;
				}
			}
			if (!done) {
				std::cout << "Case #" << tt << ": " << i*N << std::endl;
				done = true;
				break;
			}
			done = false;
		}
		if (!done) {
			std::cout << "Case #" << tt << ": INSOMNIA" << std::endl;
		}
	}
}
